optimum/neuron/accelerate/optimizer.py

```diff
@@ -40,6 +43,12 @@ def step(self, closure=None):
                 xm.optimizer_step(self.optimizer, optimizer_args=optimizer_args)
             elif self.accelerator_state.distributed_type is NeuronDistributedType.XLA_FSDP:
                 self.optimizer.step(closure)
+            elif self.accelerator_state.distributed_type is NeuronDistributedType.TENSOR_PARALLELISM:
+                xm.optimizer_step(self.optimizer, groups=parallel_state.get_data_parallel_group(as_list=True))
+                # xm.reduce_gradients(self.optimizer, groups=parallel_state.get_data_parallel_group(as_list=True))
+                # parameters = [p for group in self.optimizer.param_groups for p in group["params"]]
+                # parallel_layers.clip_grad_norm(parameters, 1.0, norm_type=2)
+                # self.optimizer.step()
```

Comment:
Do you intend to leave those comments?

Reply:
Not yet, I will later but for now it might come in handy.

Reply:
I tried with it and it still failed. Also we do not do any gradient accumulation in the tests so I am not sure?

Reply:
grad clipping is unrelated to grad_accumulation. What is the grad_clip value set by default for the non-tp case?